cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/AwsClusterSettingsStep.tsx

```diff
@@ -78,28 +78,27 @@ function AwsClusterSettingsStep(props: Partial<StepProps>) {
         } as FormAction);
     };
 
-    const osImages = (getResource(AWS_FIELDS.OS_IMAGE, awsState) || []) as AWSVirtualMachine[];
-    let initialSelectedNodeProfileId = awsState[STORE_SECTION_FORM][AWS_FIELDS.NODE_PROFILE];
-
-    if (awsState[STORE_SECTION_FORM][AWS_FIELDS.OS_IMAGE]) {
-        setValue(AWS_FIELDS.OS_IMAGE, awsState[STORE_SECTION_FORM][AWS_FIELDS.OS_IMAGE].name);
-    }
+    // NOTE: we assume that the osImages were set in the store during the credentials step
+    const osImages = getResource<AWSVirtualMachine[]>(AWS_FIELDS.OS_IMAGE, awsState) || [];
+    const initialSelectedNodeProfileId = awsState[STORE_SECTION_FORM][AWS_FIELDS.NODE_PROFILE] || nodeProfiles[0].id;
+    const [selectedNodeProfileId, setSelectedNodeProfileId] = useState(initialSelectedNodeProfileId);
 
-    if (!initialSelectedNodeProfileId) {
-        initialSelectedNodeProfileId = nodeProfiles[0].id;
-        setValue(AWS_FIELDS.NODE_PROFILE, initialSelectedNodeProfileId);
-    }
+    const onNodeProfileChange = (instanceType: string) => {
```

Comment:
I think this is one of the areas where we are using 'instanceType' but the value is actually the node profile type